draft-pauly-taps-arch.md

```diff
@@ -205,103 +205,93 @@ Beyond the basic objects, there are several high-level groups of actions that an
 
-### Objects {#objects}
+~~~~~~~~~~
+     Pre-Establishment     :       Established             : Termination
+     -----------------     :       -----------             : -----------
+                           :                     Close()   :
+     +---------------+ Initiate() +------------+ Abort()   :
+ +-->| Preconnection |----------->| Connection |---------------> Closed
+ |   +---------------+     :      +------------+ Connection:
+ |                         :      ^   ^    |     Finished  :
+ +-- Local Endpoint        :      |   |    |               :
+ |                         :      |   |    +-----+         :
+ +-- Remote Endpoint       :      |   |          |Content  :
+ |                         :      |   |Send()    |Received :
+ +-- Path Selection        :      | +---------+  |         :
+ |   Properties            :      | | Content |<-+         :
+ |                         :      | +---------+            :
+ +-- Protocol Selection    :      |                        :
+ |   Properties            :      |                        :
+ |                         :      |                        :
+ +-- Specific Protocol     :      |                        :
+ |   Properties            :      |                        :
+ |                         :      |                        :
+ |   +----------+          :      |                        :
+ +-->| Listener |-----------------+                        :
+     +----------+ Connection Received                      :
+           ^               :                               :
+           |               :                               :
+        Listen()           :                               :
+~~~~~~~~~~
+{: #fig-lifetime title="The lifetime of a connection"}
 
-Connection
+### Basic Objects {#objects}
 
->> A Connection is the fundamental object used by an application for all interaction with a peer and data transfer. It is generally capable of bi-directional communication. It has state that represents the capability of being able to send or receive content between its local and remote endpoints. This capability may be based merely on the existence of a path between the two endpoints and an application's permission to send and receive on its local endpoint; or may represent successful protocol handshakes at one or more layers that are pre-requisites to receiving content.
+* Preconnection: A Preconnection object is a representation of a potential connection. It has state that describes parameters of a Connection that might exist in the future: the Local Endpoint from which that Connection will be established, the Remote Endpoint to which it will connect, and Path Selection Properties, Protocol Selection Properties, and Specific Protocol Properties that influence the choice of transport that Connection will use. A Preconnection can be fully specified and represent a single possible Connection, or it can be partially specified such that it represents a family of possible Connections. 
```

Comment:
May be good to note that the Local Endpoint is an optional part of the pre-connection properties

Reply:
Yup - will fix